src/lib.rs

```diff
@@ -421,6 +460,28 @@ fn get_embeddings(data: &[f32], dimensions: &[usize]) -> Vec<Embedding> {
     embeddings
 }
 
+/// Get the cached onnx file from the model directory
+fn get_cached_onnx_file(model: ModelInfo, cache_dir: &PathBuf) -> Option<PathBuf> {
```

Comment:
If I'm not mistaken, we could achieve something similar using `std::fs::read_dir` recursively instead of depending on `WalkDir`.

Reply:
I think you're right - I'll take a look at switching this over

Reply:
Sorry for being slow - I've just pushed this change.